src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -1,4 +1,28 @@
-﻿namespace FSharpVSPowerTools
+﻿namespace FSharpPowerTools.Core
+
+type FileName = string
+
+type ZeroBasedPoint = int * int
+type ZeroBasedRange = int * int * int * int
+
+type CurrentLine = { Line: string; File: FileName; Range: ZeroBasedRange }
+    with
+        member x.EndLine =
+            let _, _, endLine, _ = x.Range
+            endLine
+
+[<NoComparison>]
+type PointInDocument = {
+  Point: ZeroBasedPoint
+  Line: string
+  Document: string
+  File: FileName
+}
+    with
+        member x.LineIndex = fst x.Point
+        member x.ColumnIndex = snd x.Point
+        member x.CurrentLine = {Line = x.Line; File = x.File; Range = x.LineIndex, x.ColumnIndex, x.LineIndex, x.Line.Length }
```

Comment:
This means large refactoring, and we should also clarify onebased / zerobased stuff which is error prone and make contributors spend time figuring when/where to do the conversion.

I think it makes more sense to make most of our code to be "close to FCS" and keep VS stuff only "close to VS" and convert to structures which are VS agnostic (we shouldn't cary SnapshotPoint etc. through several calls/layers).

Can we deal with this later though (I'll make a `Point` type and also use Lazy for that property)? I'd like to complete refactoring the logic for Go To Definition to move it to Core and focus on maybe other features.